src/java/src/main/java/com/dnanexus/DXSearch.java

```diff
@@ -989,14 +1022,110 @@ private T getDataObjectInstanceFromResult(FindDataObjectsResponse.Entry e) {
             return (T) dataObject;
         }
 
-
         @Override
         public Iterator<T> iterator() {
             return new ResultIterator();
         }
     }
 
     /**
+     * The page subset of data objects that matched a {@code findDataObjects} query.
+     *
+     * @param <T> data object class to be returned
+     */
+    public static class SearchPage<T extends DXDataObject> implements Iterable<T> {
```

Comment:
After last modifications the implementation of search page class became to [be quite trivial](https://github.com/storozhilov-dnanexus/dx-toolkit/blob/PTFM-15653-resolution/src/java/src/main/java/com/dnanexus/DXSearch.java#L903-L955). That is why I didn't introduce a base class for that so far, but decided to ask you first. Maybe it makes sense to do that when to implement the same pagination for executions search if we decide, that it's definitely needed?

Since I've moved **SearchPage** into **FindDataObjectsResult** it looks like naming is not an issue.

Keep me informed, please, if you think we are still having issues here.